### ISSUE TITLE
When I create new wallets in Nami, I don't get any new passphrase

### ISSUE THREAD

**Issue by luisaaagam** [Berry-Pool/nami-wallet#281]
Is this normal? I've created new wallets within my Nami account / extension, but I don't get any new passphrase. How can I retrieve the passphrases now for the new wallets?

**Comment by HeptaSean** [comment]
As far as I know, Nami only manages one wallet with only one seed phrase (but you can connect additional hardware wallets).

What you are creating with “New Account” are *not* wallets, but accounts within the same wallet, protected by the same seed phrase.

This feature is also available in Adalite and ccvault, but not in Daedalus and Yoroi. The latter will only ever see the first account (with number 0).

Should you ever need to restore/import your wallet, the subaccounts (numbers 1+) will not appear immediately, but should appear in the same order again if you do “New Account” with that restored wallet (in all of the supporting wallet apps).

**Comment by luisaaagam** [comment]
Oh wow, I didn't know this! Thanks for the help. So it's just multiple restores within the same wallet? @HeptaSean

**Comment by HeptaSean** [comment]
> So it's just multiple restores within the same wallet?

Yes, somehow. It's a way to organise your wallet. Accounts can, e.g., stake with different pools. It is a lot like having multiple wallets, but protected with the same seed phrase.

The longer explanation is that your seed phrase encodes a very large number. From this number private and public keys and addresses are derived with a specified process. What exactly is derived, is given by a path like `m/1852'/1815'/0'/1/2`. Some of the wallet apps show you these paths directly or in the details of an address. The `1852'` is the “purpose” and the `1815'` the coin type (ADA/Cardano in our case). They are pretty much constant for us, but will be different, e.g., if you use the same seed phrase (or the same hardware wallet) for Bitcoin, Ethereum, …

The `0'` is the **account**. Daedalus and Yoroi only ever use the 0, here, while Adalite, ccvault, and Nami decided to also allow you to use 1, 2, 3, … (up to now, they add/open these in exactly this order, but yet another wallet app could decide to let you freely choose a number).

The `1` is the role, `0` would be for a receive address, `1` is for a change address and `2` is for the stake key. And the last number is just a counter. All the other wallet apps use ever new numbers, here, when creating transactions or proposing you receive addresses, but Nami only ever uses `/0/0` for the last two components. For stake keys, only the `0` is used (but that's also only a convention and wallet apps could decide differently).

The Cardano network itself does not know about all this. It's all happening inside the wallet apps (or inside the hardware wallets). It just knows about public keys and addresses and can verify if a transaction was signed by the right private keys. If they were derived by this process (and by which options in this process) or if they were freely diced without using a seed phrase at all, is not known and irrelevant to the nodes in the network.

If this answers more than you ever wanted to know, please consider closing this issue, so that the Nami people don't have to deal with so much stuff, when they come to going through all this. Thank you!
